metrics/metrics/src/main/java/io/helidon/metrics/RegistryFactory.java

```diff
@@ -74,20 +80,61 @@ static RegistryFactory create() {
      * @return supplier of registry factory (to bind as late as possible)
      */
     public static Supplier<RegistryFactory> getRegistryFactory() {
-        return () -> staticInstance;
+        return () -> {
```

Comment:
Could this simply invoke `getInstance()` for the caller, rather than throw an exception? 

Doing so might mask the case where the developer intended to use config but forgot to. But if the developer had invoked `getInstance()` explicitly instead of `getInstance(Config)` s/he would have to diagnose the same sort of problem.

Reply:
Refactored a bit more - now we mostly use a singleton instance, with support for customized RegistryFactory (and MetricSupport) - if for any reason somebody wants more than one RegistryFactory in their application accessible on different endpoints